src/System.Net.Mail/tests/Functional/CredentialTest.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Mail.Functional.Tests
+{
+    public class CredentialTest
+    {
+        private ICredentialsByHost GetTransportCredentials(SmtpClient client)
+        {
+            Type smtpTransportType = (typeof(SmtpClient)).Assembly.GetType("System.Net.Mail.SmtpTransport");
+
+            var transport = typeof(SmtpClient)
+                .GetField("_transport", BindingFlags.NonPublic | BindingFlags.GetField | BindingFlags.Instance)
+                .GetValue(client);
+
+            var transportCredentials = smtpTransportType
+                .GetProperty("Credentials", BindingFlags.NonPublic | BindingFlags.GetProperty | BindingFlags.Instance)
+                .GetValue(transport);
+
+            return (ICredentialsByHost) transportCredentials;
+        }
+
+        [Fact]
+        public void Credentials_Unset_Null()
+        {
+            SmtpClient client = new SmtpClient();
+
+            ICredentialsByHost transportCredentials = GetTransportCredentials(client);
+
+            Assert.Null(client.Credentials);
+
+            // Verify that transport credentials are correct
+            Assert.Null(transportCredentials);
+        }
+
+        [Fact]
+        public void DefaultCredentials_True_DefaultCredentials()
+        {
+            NetworkCredential expectedCredentials = CredentialCache.DefaultNetworkCredentials;
+
+            SmtpClient client = new SmtpClient();
+            client.UseDefaultCredentials = true;
+
+            ICredentialsByHost transportCredentials = GetTransportCredentials(client);
+
+            Assert.Equal(expectedCredentials, client.Credentials);
+
+            // Verify that transport credentials are correct
+            Assert.Equal(expectedCredentials, transportCredentials);
+        }
+
+        [Fact]
+        public void UseDefaultCredentials_False_Null()
+        {
+            SmtpClient client = new SmtpClient();
+            client.UseDefaultCredentials = false;
+
+            ICredentialsByHost transportCredentials = GetTransportCredentials(client);
+
+            Assert.Null(client.Credentials);
+
+            // Verify that transport credentials are correct
+            Assert.Null(transportCredentials);
+        }
+
+        [Fact]
+        public void Credentials_UseDefaultCredentialsSetFalseBeforeCredentials_Credentials()
+        {
+            string userName = "user";
```

Comment:
These variables for 'userName' and 'password' are repeated in several test methods. Also, using source code of the form:

```c#
```

will cause our static analysis tools to falsely flag this as a security problem (hard-coding passwords in code). Yes, it is a dummy password.

To avoid these problems, please:

1. create constants in the test class for these things to reduce duplicating code in the test methods
2. use something like `Guid` to set the dummy password

```c#
private readonly string UserName = "user";
private readonly string Password = Guid.NewGuid().ToString();
```